code/extensions/che-activity-tracker/src/activity-tracker-service.ts

```diff
@@ -0,0 +1,81 @@
+/**********************************************************************
+ * Copyright (c) 2019-2022 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ ***********************************************************************/
+
+/**
+ * Receives activity updates and sends reset inactivity requests to the che-exec /activity/tick endpoint.
+ * To avoid duplicate requests may send requests periodically. This means
+ * that, in the worst case, it might keep user's workspace alive for a longer period of time.
+ */
+export class ActivityTrackerServiceImpl {
```

Comment:
maybe without the `Impl` suffix, as there's no interface

Reply:
Sounds good, updateds